### PR TITLE
Release v0.9.0

### PR DESCRIPTION
**Pull request by reyoung**
* Merge release/v0.9.0 to master

## Release Note

TODO @backyes @qingqing01 @hedaoyuan @gangliao  Please add release note here.

## QA Testing

- [x] Yang WenBo
- [x] Qi Bin
- [x] Xu jun

### CODE REVIEW

**Comment by gangliao** [comment]
- [x] Add bilinear interpolation layer for image classification
- [x] Add code coverage and [coveralls](https://coveralls.io/github/baidu/Paddle)
- [x] Support MAC OS Sierra
- [x] Fix some bugs, for instance,  default value of`WITH_AVX` and `AVX_FLAG` in CMakeLists.txt

**Comment by backyes** [comment]
* Bugfix: link zlib into Paddle if linking Paddle statically.
* Support link RDMA lib with cmake system
* Bugfix: Include glog header file if necessary
* Bugfix: gently go forward if empty parameter server was found
* Bugfix: Add input sparse data check for sparse layer at runtime

**Comment by hedaoyuan** [comment]
- [x] Bug fix for sparse matrix multiplication
- [x] Fix floating-point overflow problem of tanh
- [x] Fix some nvcc compile options

**Comment by luotao1** [comment]
1. python接口增强： 
- cost layer增加weight layer的input接口
- nce layer的支持
- 修复block_expand_layer的接口问题
2. 双层RNN文档和单测的完善
3. 支持maxout layer
4. --save dir 命令行参数可以支持多种路径形式

**Comment by qingqing01** [comment]
New Features中1-2分别是qijun、wangyang的。

New Features:
1. Support spatial pyramid-pool layer.
2. Support de-convolution layer.
3. Support rectangle padding, stride, window and input for Pooling Operation.	
4. Add `—job=time` in trainer, which can be used to print time info without compiler option ` -WITH_TIMER=ON`.

Improvement:
1. Refine convolution layer to speedup and reduce GPU memory. 
2. fix some bugs.
3. improve some demo.

**Comment by reyoung** [comment]
* Add ubuntu deb package build scripts.
	* use docker to build ubuntu 14.04 package
* Add travis for mac osx, enable swig unittest in travis
* Add FAQ docs
* Fix a bug in yield dictionary in DataProvider
* Speed up DataProvider
* Add Bidi-LSTM and DB-LSTM to quick start demo @alvations
* Make Paddle use git-flow branching model.
* Add concepts docs
* Fix SRL hang when exit.
* Add usage track scripts.

**Comment by reyoung** [comment]
@wangkuiyi All things done for v0.9.0. Should it be merged to master release?

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Release v0.9.0
+
+## New Features:
+
+* Add some layers to Paddle
```
Comment (wangkuiyi) [inline RELEASE.md]: Add some layers to Paddle ==> New Layers

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  * bilinear interpolation layer.
+  * spatial pyramid-pool layer.
+  * de-convolution layer.
+  * maxout layer.
+* Support rectangle padding, stride, window and input for Pooling Operation.
+* Add —job=time in trainer, which can be used to print time info without compiler option -WITH_TIMER=ON.
+* Support Mac OS X Sierra by source code.
```
Comment (wangkuiyi) [inline RELEASE.md]: Mac OS X Sierra 确实可以build吗？我在Yosemite上build是过不了，因为google test的源码不够符合C++11的规范。如果我们号称支持Mac，就得做到没问题。如果我们稳定思路是Docker为唯一官方支持方式，这里最好不claim支持mac。

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+* Expose cost_weight/nce_layer in `trainer_config_helpers`
+* Add FAQ, concepts, h-rnn docs.
+* Add Bidi-LSTM and DB-LSTM to quick start demo @alvations
+* Add usage track scripts.
+
+## Improvements
+
+* Add travis-ci for macos. Enable swig unittest in travis. Skip travis-ci when only docs are changed.
```
Comment (wangkuiyi) [inline RELEASE.md]: travis-ci ==> Travis-CI
macos ==> Mac OS X

Comment (reyoung) [inline RELEASE.md]: Fix in another PR. #652

File: RELEASE.md (hunk 2 above)
Comment (gangliao) [inline RELEASE.md]: 1. 我们的 Travis https://travis-ci.org/  一直再跑Mac OS X的 build and test， 一直没问题 。
2. 我已经使用Mac Sierra两个月了，一直没问题。用户Sierra build也没有问题：https://github.com/PaddlePaddle/Paddle/issues/551
3. 我认为build本身是没问题的，关键在于build 源码比较费心，容易出错。

我建议在` Support Mac OS X Sierra by source code ` 后面加上 
`but [docker image](http://www.paddlepaddle.org/doc/build/docker_install.html) is officially recommended to install PaddlePaddle.`

Comment (gangliao) [inline RELEASE.md]: 费心，指的是各种环境依赖容易导致源码编译失败。

Sierra         == OS X 10.12 == My laptop  => OK
El Capitan  == OS X 10.11 == Travis OS X default  => OK
Yosemite   == OS X 10.10 == ? 之所以编译不过，原因肯定在于XCode自带的Clang版本不同，一直没有测试。

我们Travis OS X支持的各种版本其实很全，后期可以考虑加上。
https://docs.travis-ci.com/user/osx-ci-environment/

Comment (wangkuiyi) [inline RELEASE.md]: 如果很希望这个工作进入release note，请加上你使用的Xcode的版本吧。

如果写入release note了，就不要注明docker是唯一支持的模式了。

File: RELEASE.md (hunk 3 above)
Comment (wangkuiyi) [inline RELEASE.md]: Approved.

**Pull request merged**